src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/TagHelperSemanticSpanVisitor.cs

```diff
@@ -0,0 +1,217 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Language.Components;
+using Microsoft.AspNetCore.Razor.Language.Syntax;
+using Microsoft.AspNetCore.Razor.LanguageServer.Semantic.Models;
+using Microsoft.VisualStudio.Editor.Razor;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Semantic
+{
+    internal class TagHelperSemanticSpanVisitor : SyntaxWalker
```

Comment:
Sorry I know I asked you to rename this and suggested `TagHelperSemanticSpanVisitor` but given the `SemanticRange`ness of the API it should probably be `TagHelperSemanticRangeVisitor`